src/main/scala/mesosphere/marathon/DeprecatedFeatures.scala

```diff
@@ -0,0 +1,49 @@
+package mesosphere.marathon
+
+import com.typesafe.scalalogging.StrictLogging
+
+object DeprecatedFeatures extends StrictLogging {
+  case class DeprecatedFeature(
+      key: String,
+      description: String,
+      warnVersion: String,
+      removeVersion: String
+  )
+
+  val syncProxy = DeprecatedFeature(
+    "sync_proxy",
+    description = "Old, blocking IO implementation for leader proxy used by Marathon standby instances.",
+    warnVersion = "1.7.0",
+    removeVersion = "1.8.0")
+
+  val all = Seq(syncProxy)
+
+  private val versionOrdering: Ordering[Iterable[Int]] = implicitly[Ordering[Iterable[Int]]]
+
+  private def simpleVersionParse(ver: String): List[Int] =
```

Comment:
What do you think about a Semver case class? That's easier to handle, see eg the [StorageVersion](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/storage/migration/Migration.scala#L252) for migrations.

Reply:
Oh geez yeah that would be way easier to use

Reply:
Hmm, that class doesn't look entirely usable at this point; were you thinking of importing the SemVer class from our CI and integrating it, etc?

Reply:
We could do that or define another here. The ordering is much easier to understand than the current.